addons/point_of_sale/models/pos_order.py

```diff
@@ -90,6 +90,9 @@ def _process_order(self, order, existing_order):
         """
         draft = True if order.get('state') == 'draft' else False
         pos_session = self.env['pos.session'].browse(order['session_id'])
+        if not pos_session and existing_order:
+            order['session_id'] = existing_order.session_id.id
```

Comment:
```suggestion
            pos_session = existing_order.session_id.id
```
@davidmonnom I think we should do this.